frontend/js/components/TableSettledRow.js

```diff
@@ -31,24 +32,24 @@ const Image = styled.img`
 
 class TableSettledRow extends Component {
   componentDidMount() {
-    const { setSettledBattles } = this.props;
-    setSettledBattles();
+    const { fetchSettledBattles } = this.props;
+    fetchSettledBattles();
   }
 
   render() {
-    const { battles, session } = this.props;
-    if (!battles) return <>Loading</>;
-    const battleList = Object.values(battles);
+    const { battlesResult, battleList, session } = this.props;
+    if (!battlesResult) return <>Loading</>;
     return (
       <div>
-        {battleList.map((battle) => {
-          const { id, created, opponent, creator, winner } = battle;
+        {battlesResult.map((id) => {
+          const battle = battleList[id];
+          const { created, opponent, creator, winner } = battle;
```

Comment:
Ideally this should be an own component. That receives an Id and fetches the battle. This component is doing too much: Handling the list and the individual items. In this case is fine because the list logic is small but it could be bigger, so we need to Separate the concerns. See https://en.wikipedia.org/wiki/Separation_of_concerns , https://deviq.com/separation-of-concerns/ or similar articles.

Reply:
Thanks for the links, I'm getting familiar with this concept in React. I think I was able to solve it, if the change I made was not the ideal solution feel free to say it!